src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/redundantthis/InputRedundantThisEnumInnerClasses.java

```diff
@@ -0,0 +1,53 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.redundantthis;
+
+public class InputRedundantThisEnumInnerClasses {
+    int i;
+    int j;
+    void method1() {
+        i = 3;
+    }
+
+    void method2(int j) {
+        i++;
+        this.j = j; // no violation
+        method1();
+        try {
+            this.method1(); // violation
+        }
+        catch (RuntimeException e) {
+            e.toString();
+        }
+        this.i--; // violation
+
+        Integer.toString(10);
+    }
+
+    void method3()
+    {
+        i = 3;
+    }
+
+    void method4() {
+        this.method3(); // violation
+        method3();
+    }
+}
+
+//  enum
+enum MyEnum
+{
```

Comment:
please make formatting consistent and make lcurly on previous line

Reply:
Done.